src/actions/makeAuction.ts

```diff
@@ -0,0 +1,56 @@
+import { Transaction } from '@metaplex-foundation/mpl-core';
+import {
+  Auction,
+  AuctionExtended,
+  CreateAuction,
+  CreateAuctionArgs,
+} from '@metaplex-foundation/mpl-auction';
+import { PublicKey, TransactionSignature } from '@solana/web3.js';
+
+import { Wallet } from '../wallet';
+import { Connection } from '../Connection';
+import { sendTransaction } from './transactions';
+
+interface MakeAuctionParams {
+  connection: Connection;
+  wallet: Wallet;
+  vault: PublicKey;
+  auctionSettings: CreateAuctionArgs;
+}
+
+interface MakeAuctionResponse {
+  txId: TransactionSignature;
+  auction: PublicKey;
+}
+
+export const makeAuction = async ({
```

Comment:
can we combine create a vault and create auction in this action?
because a vault doesn't make sense without an auction.
In my opinion, the API should look like this:
`createAuction(nft, some args)`
`startAuction(action)`

Reply:
vault creation in scope of new auction has been moved to makeAuction
createVault still exists as a separate action for testing purposes

Reply:
@kurpav please have a look at https://github.com/metaplex-foundation/metaplex/blob/master/js/packages/web/src/actions/createAuctionManager.ts if it is satisfying your vision of API look. 
If yes, I will revert changes related to creation of new vault in makeAuction action

Reply:
Yes, I think we should have a similar API. You can split it into small pieces and in the end, just combine it to have the action `createAuction` which will do all the magic under the hood.

Reply:
I see, but from my point of view we should keep makeAuction as it is right now as auction does not make sense without auctionManager. For this guy we have https://github.com/metaplex-foundation/js/issues/96 and I believe we should combine all of the chunks in scope of it